sdk/storage/blobs/block_blob_client.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+namespace Azure { namespace Storage {
+
+class BlockBlobClient : public BlobClient
+{
+public:
+    // connection string
+    BlockBlobClient(const std::string& connectionString, std::string containerName, std::string blobName, BlobClientOptions options) : BlobClient(connectionString, std::move(containerName), std::move(blobName), std::move(options)) {}
+
+    // shared key auth
+    BlockBlobClient(Azure::Core::Http::Uri uri, Azure::Storage::Common::StorageSharedKeyCredential credential, BlobClientOptions options) : BlobClient(std::move(uri), std::move(credential), std::move(options)) {}
+
+    // token auth
+    BlockBlobClient(Azure::Core::Http::Uri uri, Azure::Core::TokenCredential credential, BlobClientOptions options) : BlobClient(std::move(uri), std::move(credential), std::move(options)) {}
+
+    // anonymous/SAS/customized pipeline auth
+    BlockBlobClient(Azure::Core::Http::Uri uri, BlobClientOptions options) : BlobClient(std::move(uri), std::move(options)) {}
+
+    Azure::Core::Http::Response<BlockInfo> StageBlock(const std::string& blockId, std::istream& content, const StageBlockOptions& options = StageBlockOptions());
```

Comment:
>StageBlock [](start = 43, length = 10)

Stage is a strange verb. We should be first align with REST, and consider the align with .NET only if we cannot be align with REST.

Reply:
`stage` aligns with .Net and Javascript.

Reply:
But I'm also okay with `PutBlock` and `PutBlockList`.

Reply:
Let's leave it an open discussion. I'll log it into our meeting minutes.